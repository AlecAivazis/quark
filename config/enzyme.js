// external imports
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import browserEnv from 'browser-env'

// add a request animation frame polyfill
import raf from 'raf'
raf.polyfill()

// enable the react 16 adapter for enzyme
Enzyme.configure({ adapter: new Adapter() })

// add various browser mocks
browserEnv()
