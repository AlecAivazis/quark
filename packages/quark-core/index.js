export * from './components';
import * as _styles from './styles';
export { _styles as styles };