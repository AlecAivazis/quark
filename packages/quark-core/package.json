{
    "name": "quark-core",
    "version": "1.0.6",
    "module": "index.js",
    "main": "index.js",
    "dependencies": {
        "react-native-web": "0.3.4",
        "svgs": "^3.1.1"
    },
    "peerDependencies": {
        "react": "^16.3.0"
    }
}
