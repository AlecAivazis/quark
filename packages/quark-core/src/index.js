// @flow
export * from './components'
export * as styles from './styles'
