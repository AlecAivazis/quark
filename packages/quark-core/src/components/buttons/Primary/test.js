// external imports
import React from 'react'
import { Text } from 'react-native'
import { mount } from 'enzyme'
// local imports
import { PrimaryButton } from 'quark-core'

it('renders', () => {
    expect(
        mount(
            <PrimaryButton>
                <Text>hello</Text>
            </PrimaryButton>
        )
    ).toMatchSnapshot()
})

it('accepts raw text as a child', () => {
    expect(mount(<PrimaryButton>hello</PrimaryButton>)).toMatchSnapshot()
})
