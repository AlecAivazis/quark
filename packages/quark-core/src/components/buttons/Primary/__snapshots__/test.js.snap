// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accepts raw text as a child 1`] = `
<Button>
  <GetTheme>
    <ButtonWithText
      activeColor="#2B598E"
      defaultColor="#4A90E2"
      disabledColor="#EAEAEE"
      size="medium"
      textColor="#ffffff"
    >
      <BaseButton
        activeColor="#2B598E"
        constrainSize={true}
        defaultColor="#4A90E2"
        disabledColor="#EAEAEE"
        size="medium"
      >
        <TouchableWithoutFeedback
          accessible={true}
          onPressIn={[Function]}
          onPressOut={[Function]}
        >
          <AnimatedComponent
            accessible={true}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "animationDuration": "0.75s",
                "animationIterationCount": "infinite",
                "animationName": "rn-ActivityIndicator-animation",
                "animationTimingFunction": "linear",
                "backgroundColor": "rgba(74, 144, 226, 1)",
                "backgroundSize": "cover",
                "color": "#fff",
                "fontWeight": "500",
                "height": 36,
                "padding": 8,
                "textAlign": "center",
                "textTransform": "uppercase",
                "width": 36,
              }
            }
          >
            <View
              accessible={true}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(74, 144, 226, 1)",
                  "borderRadius": 2,
                  "cursor": "pointer",
                  "height": 40,
                  "justifyContent": "center",
                  "maxWidth": 240,
                  "minWidth": 120,
                  "overflow": "hidden",
                  "touchAction": "manipulation",
                }
              }
            >
              <div
                className="rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "WebkitAlignItems": "center",
                    "WebkitBoxAlign": "center",
                    "WebkitBoxPack": "center",
                    "WebkitJustifyContent": "center",
                    "alignItems": "center",
                    "backgroundColor": "rgba(74, 144, 226, 1)",
                    "borderBottomLeftRadius": "2px",
                    "borderBottomRightRadius": "2px",
                    "borderTopLeftRadius": "2px",
                    "borderTopRightRadius": "2px",
                    "cursor": "pointer",
                    "height": "40px",
                    "justifyContent": "center",
                    "maxWidth": "240px",
                    "minWidth": "120px",
                    "msFlexAlign": "center",
                    "msFlexPack": "center",
                    "msTouchAction": "manipulation",
                    "overflowX": "hidden",
                    "overflowY": "hidden",
                    "touchAction": "manipulation",
                  }
                }
                tabIndex="0"
              >
                <Text
                  style={
                    Object {
                      "color": "#ffffff",
                      "textStyle": undefined,
                    }
                  }
                >
                  <InnerText
                    style={
                      Object {
                        "color": "#ffffff",
                        "textStyle": undefined,
                      }
                    }
                  >
                    <GetTheme>
                      <Text
                        style={
                          Object {
                            "color": "#ffffff",
                            "textStyle": undefined,
                          }
                        }
                      >
                        <div
                          className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-fontFamily-14xgk7a rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
                          dir="auto"
                          style={
                            Object {
                              "color": "rgba(255,255,255,1)",
                            }
                          }
                        >
                          hello
                        </div>
                      </Text>
                    </GetTheme>
                  </InnerText>
                </Text>
              </div>
            </View>
          </AnimatedComponent>
        </TouchableWithoutFeedback>
      </BaseButton>
    </ButtonWithText>
  </GetTheme>
</Button>
`;

exports[`renders 1`] = `
<Button>
  <GetTheme>
    <ButtonWithText
      activeColor="#2B598E"
      defaultColor="#4A90E2"
      disabledColor="#EAEAEE"
      size="medium"
      textColor="#ffffff"
    >
      <BaseButton
        activeColor="#2B598E"
        constrainSize={true}
        defaultColor="#4A90E2"
        disabledColor="#EAEAEE"
        size="medium"
      >
        <TouchableWithoutFeedback
          accessible={true}
          onPressIn={[Function]}
          onPressOut={[Function]}
        >
          <AnimatedComponent
            accessible={true}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "animationDuration": "0.75s",
                "animationIterationCount": "infinite",
                "animationName": "rn-ActivityIndicator-animation",
                "animationTimingFunction": "linear",
                "backgroundColor": "rgba(74, 144, 226, 1)",
                "backgroundSize": "cover",
                "color": "#fff",
                "fontWeight": "500",
                "height": 36,
                "padding": 8,
                "textAlign": "center",
                "textTransform": "uppercase",
                "width": 36,
              }
            }
          >
            <View
              accessible={true}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgba(74, 144, 226, 1)",
                  "borderRadius": 2,
                  "cursor": "pointer",
                  "height": 40,
                  "justifyContent": "center",
                  "maxWidth": 240,
                  "minWidth": 120,
                  "overflow": "hidden",
                  "touchAction": "manipulation",
                }
              }
            >
              <div
                className="rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "WebkitAlignItems": "center",
                    "WebkitBoxAlign": "center",
                    "WebkitBoxPack": "center",
                    "WebkitJustifyContent": "center",
                    "alignItems": "center",
                    "backgroundColor": "rgba(74, 144, 226, 1)",
                    "borderBottomLeftRadius": "2px",
                    "borderBottomRightRadius": "2px",
                    "borderTopLeftRadius": "2px",
                    "borderTopRightRadius": "2px",
                    "cursor": "pointer",
                    "height": "40px",
                    "justifyContent": "center",
                    "maxWidth": "240px",
                    "minWidth": "120px",
                    "msFlexAlign": "center",
                    "msFlexPack": "center",
                    "msTouchAction": "manipulation",
                    "overflowX": "hidden",
                    "overflowY": "hidden",
                    "touchAction": "manipulation",
                  }
                }
                tabIndex="0"
              >
                <Text
                  style={
                    Object {
                      "color": "#ffffff",
                      "textStyle": undefined,
                    }
                  }
                >
                  <InnerText
                    style={
                      Object {
                        "color": "#ffffff",
                        "textStyle": undefined,
                      }
                    }
                  >
                    <GetTheme>
                      <Text
                        style={
                          Object {
                            "color": "#ffffff",
                            "textStyle": undefined,
                          }
                        }
                      >
                        <div
                          className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-fontFamily-14xgk7a rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
                          dir="auto"
                          style={
                            Object {
                              "color": "rgba(255,255,255,1)",
                            }
                          }
                        >
                          <Text>
                            <span
                              className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-fontFamily-poiln3 rn-fontSize-7cikom rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-irrty rn-wordWrap-qvutc0"
                              dir="auto"
                            >
                              hello
                            </span>
                          </Text>
                        </div>
                      </Text>
                    </GetTheme>
                  </InnerText>
                </Text>
              </div>
            </View>
          </AnimatedComponent>
        </TouchableWithoutFeedback>
      </BaseButton>
    </ButtonWithText>
  </GetTheme>
</Button>
`;
